### ISSUE TITLE
Greedy feature request: Support changing manifest

### ISSUE THREAD

**Issue by Keno** [timholy/Revise.jl#238]
It often happens that I find a bug in a package that one of my packages depends on, so I `dev` it, but now I have the wrong package loaded. Often it's actually the same package as I already have loaded (because `dev` just gave me the same contents but as a git repo), but in a different place. It would be nice if Revise could switch over the module to the `dev`'ed copy.

**Comment by timholy** [comment]
Fixed in #217.

```julia
julia> Example.f()
ERROR: MethodError: no method matching f()
Stacktrace:
 [1] top-level scope at none:0

(v1) pkg> dev Example
[ Info: Path `/home/tim/.julia/dev/Example` exists and looks like the correct package, using existing path
 Resolving package versions...
  Updating `~/.julia/environments/v1/Project.toml`
  [7876af07] ↑ Example v0.5.1 ⇒ v0.5.1+ [`~/.julia/dev/Example`]
  Updating `~/.julia/environments/v1/Manifest.toml`
  [7876af07] ↑ Example v0.5.1 ⇒ v0.5.1+ [`~/.julia/dev/Example`]

julia> Example.f()
π = 3.1415926535897...

(v1) pkg> free Example
 Resolving package versions...
  Updating `~/.julia/environments/v1/Project.toml`
  [7876af07] ↓ Example v0.5.1+ [`~/.julia/dev/Example`] ⇒ v0.5.1
  Updating `~/.julia/environments/v1/Manifest.toml`
  [7876af07] ↓ Example v0.5.1+ [`~/.julia/dev/Example`] ⇒ v0.5.1

julia> Example.f()
ERROR: MethodError: no method matching f()
Stacktrace:
 [1] top-level scope at none:0

(v1) pkg> dev Example
[ Info: Path `/home/tim/.julia/dev/Example` exists and looks like the correct package, using existing path
 Resolving package versions...
  Updating `~/.julia/environments/v1/Project.toml`
  [7876af07] ↑ Example v0.5.1 ⇒ v0.5.1+ [`~/.julia/dev/Example`]
  Updating `~/.julia/environments/v1/Manifest.toml`
  [7876af07] ↑ Example v0.5.1 ⇒ v0.5.1+ [`~/.julia/dev/Example`]

julia> Example.f()
π = 3.1415926535897...

```

Prediction: you have some old packages lying around, installed prior to the days when `add`ed packages were checked out with read-only flags. Do a `Pkg.gc()` and then see if it works. You might have to forcibly remove and re-add some things.

**Comment by timholy** [comment]
Closing pending a MWE that illustrates a problem with the solution currently in place.
